tex/chapters/07-ApplyingModel.tex

```diff
@@ -0,0 +1,197 @@
+
+\section{Applying Model to Axelrod-Python}
+
+The final section of this report addresses the problem of determining when strategies in Axelrod-Python are identical.
+The assumption made in Section \ref{sec:training_model} must now be ignored.
+It stated: \textit{Strategies with different names are assumed to be different}, but this is not necessarily the case.
+
+In order to determine whether any strategies in Axelrod-Python were identical, all of them were entered into an Ashlock Tournament.
+The corresponding results table where all strategies are compared (as described in Section \ref{sec:ashlock_tourn}) was then computed, and the model was applied to this.
+A new table could then be produced containing pairs of strategies that the model predicted were equivalent.
+
+In Figure \ref{fig:matrix_similarity}, this table is presented as a Matrix Plot.
+Note that the leading diagonal represents strategies being equivalent to themselves, also the plot is symmetric, because if strategy $A$ is similar to strategy $B$ then strategy $B$ is similar to strategy $A$.
+
+\begin{figure}[htbp!]
+    \centering
+    \includegraphics[width=0.8\linewidth]{../img/ML/similarity_heatmap.png}
+    \caption{Matrix Plot of similarity}
+    \label{fig:matrix_similarity}
+\end{figure}
+
+However this plot does not provide much insight.
+Instead it can be presented as Network Graph, as in Figure \ref{fig:overall_neighbourhoods}.
+Here nodes represent strategies, and edges link nodes that the model predicted to be equivalent.
+The graph is undirected for the same reason that Figure \ref{fig:matrix_similarity} is symmetric.
+For clarity, strategies that are only similar to themselves have been omitted.
+
+\begin{figure}[htbp!]
+    \centering
+    \includegraphics[width=0.8\linewidth]{../img/neighbourhoods/overall.png}
+    \caption{Network Graph of strategies that the model had deemed equivalent}
+    \label{fig:overall_neighbourhoods}
+\end{figure}
+
+In Figure \ref{fig:sep_neighbourhoods}, all of the disjoint neighbourhoods from Figure \ref{fig:overall_neighbourhoods} have been separated out and some of these will now be explored.
+Figures \ref{fig:nbh_soft_gbm} and \ref{fig:nbh_hard_gbm} are perfect examples of the model exactly as expected.
+Figure \ref{fig:nbh_soft_gbm} shows that the model predicted several SoftGoByMajority strategies to be equivalent to each other.
+The only difference between them was the memory depth.
+Again in Figure \ref{fig:nbh_hard_gbm} the only difference between the various HardGoByMajority strategies was the memory depth.
+However, the model was able to distinguish between all Hard and Soft version of GoByMajority, hence the two networks are disjoint.
+
+In Figure \ref{fig:nbh_alt_cd} the strategies Alternator and Cycler(DC) are deemed equivalent.
+Again this is encouraging as the strategies are identical except that one begins with a Cooperation and the other begins with a Defection.
```

Comment:
It's not that encouraging, they are fundamentally different right? This is more that it's understandable.